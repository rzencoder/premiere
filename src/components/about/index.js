import React from "react";
import { Container, Item, Divider, Text, Title, Icon } from "./styles/about";

export default function About({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

About.Item = function AboutItem({ children, ...restProps }) {
  return <Item {...restProps}>{children}</Item>;
};

About.Divider = function AboutDivider({ children, ...restProps }) {
  return <Divider {...restProps} />;
};

About.Title = function AboutTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

About.Text = function AboutText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

About.Icon = function AboutIcon({ src, ...restProps }) {
  return <Icon {...restProps} src={src} />;
};
