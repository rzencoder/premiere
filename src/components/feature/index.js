import React from "react";
import {
  Background,
  Container,
  Overlay,
  TextContainer,
  Title,
  Text,
  Button,
  Details,
  SubText,
  Rating,
  AgeRating,
  Year,
} from "./styles/feature";

export default function Feature({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Feature.Background = function FeatureBackground({ ...restProps }) {
  return <Background {...restProps} />;
};

Feature.Overlay = function FeatureOverlay({ ...restProps }) {
  return <Overlay {...restProps} />;
};

Feature.TextContainer = function FeatureTextContainer({
  children,
  ...restProps
}) {
  return <TextContainer {...restProps}>{children}</TextContainer>;
};

Feature.Title = function FeatureTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Feature.Text = function FeatureText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Feature.Button = function FeatureButton({ children, ...restProps }) {
  return <Button {...restProps}>{children}</Button>;
};

Feature.Details = function FeatureDetails({ children, ...restProps }) {
  return <Details {...restProps}>{children}</Details>;
};

Feature.SubText = function FeatureSubText({ children, ...restProps }) {
  return <SubText {...restProps}>{children}</SubText>;
};

Feature.AgeRating = function FeatureAgeRating({ ...restProps }) {
  return <AgeRating {...restProps} />;
};

Feature.Year = function FeatureYear({ children, ...restProps }) {
  return (
    <Year {...restProps}>
      <div>{children}</div>
    </Year>
  );
};

Feature.Rating = function FeatureRating({ children, ...restProps }) {
  return (
    <Rating {...restProps}>
      <div>{children}</div>
    </Rating>
  );
};
