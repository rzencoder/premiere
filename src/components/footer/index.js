import React from "react";
import {
  Container,
  Text,
  Subtitle,
  Section,
  Link,
  Divider,
  Column,
  Break,
} from "./styles/footer";

export default function Footer({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Footer.Text = function FooterText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Footer.Subtitle = function FooterSubtitle({ children, ...restProps }) {
  return <Subtitle {...restProps}>{children}</Subtitle>;
};

Footer.Link = function FooterLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>;
};

Footer.Divider = function FooterDivider({ ...restProps }) {
  return <Divider {...restProps} />;
};

Footer.Section = function FooterSection({ children, ...restProps }) {
  return <Section {...restProps}>{children}</Section>;
};

Footer.Column = function FooterColumn({ children, ...restProps }) {
  return <Column {...restProps}>{children}</Column>;
};

Footer.Break = function FooterBreak({ ...restProps }) {
  return <Break {...restProps} />;
};
