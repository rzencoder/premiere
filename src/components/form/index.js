import React from "react";
import { Link as ReactRouterLink } from "react-router-dom";
import {
  Container,
  Title,
  Content,
  Label,
  Input,
  Submit,
  Frame,
  Text,
  Link,
  Error,
} from "./styles/form";

export default function Form({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Form.Frame = function FormFrame({ children, ...restProps }) {
  return (
    <Frame {...restProps}>
      <div>{children}</div>
    </Frame>
  );
};

Form.Title = function FormTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Form.Content = function FormContent({ children, ...restProps }) {
  return <Content {...restProps}>{children}</Content>;
};

Form.Label = function FormLabel({ children, ...restProps }) {
  return <Label {...restProps}>{children}</Label>;
};

Form.Input = function FormInput({ ...restProps }) {
  return <Input {...restProps} />;
};

Form.Submit = function FormSubmit({ children, ...restProps }) {
  return <Submit {...restProps}>{children}</Submit>;
};

Form.Text = function FormText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Form.Link = function FormLink({ to, children, ...restProps }) {
  return (
    <ReactRouterLink to={to}>
      <Link {...restProps}>{children}</Link>
    </ReactRouterLink>
  );
};

Form.Error = function FormError({ children, ...restProps }) {
  return <Error {...restProps}>{children}</Error>;
};
