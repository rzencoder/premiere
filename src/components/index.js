export { default as Navigation } from "./navigation";
export { default as Jumbotron } from "./jumbotron";
export { default as About } from "./about";
export { default as Footer } from "./footer";
export { default as Form } from "./form";
