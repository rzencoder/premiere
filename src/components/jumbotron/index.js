import React from "react";
import {
  Background,
  Container,
  Overlay,
  Group,
  Title,
  Text,
  CallToAction,
} from "./styles/jumbotron";
import { Link as ReactRouterLink } from "react-router-dom";

export default function Jumbotron({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Jumbotron.Overlay = function JumbotronOverlay({ ...restProps }) {
  return <Overlay {...restProps} />;
};

Jumbotron.Group = function JumbotronGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>;
};
Jumbotron.Title = function JumbotronTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Jumbotron.Text = function JumbotronText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Jumbotron.CallToAction = function JumbotronCallToAction({
  to,
  children,
  ...restProps
}) {
  return (
    <ReactRouterLink to={to}>
      <CallToAction {...restProps}>{children}</CallToAction>
    </ReactRouterLink>
  );
};

Jumbotron.Background = function JumbotronBackground({ ...restProps }) {
  return <Background {...restProps} />;
};
