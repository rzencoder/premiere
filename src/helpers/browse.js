export function filterGenres(data) {
  const genres = [...new Set(data.map((item) => item.genre))];
  const genresSet = [...new Set(genres.flat())].sort();
  return ["All", ...genresSet];
}

export function displayAgeRating(rating) {
  const path = "../images/icons/";
  switch (rating) {
    case "PG-13":
      return `${path}12a.png`;
    case "PG":
    case "TV-PG":
      return `${path}pg.png`;
    case "R":
    case "TV-14":
      return `${path}15.png`;
    case "NC-17":
    case "TV-MA":
      return `${path}18.png`;
    case "G":
      return `${path}universal.png`;
    default:
      return null;
  }
}
const ratingColors = [
  "#227b08",
  "#6a9d0f",
  "#b5cc13",
  "#dfd31d",
  "#d59110",
  "#dd7122",
  "#e12121",
];

export function getRatingColor(data) {
  const rating = parseFloat(data);
  switch (true) {
    case rating > 8.5:
      return ratingColors[0];
    case rating > 7.9:
      return ratingColors[1];
    case rating > 7.3:
      return ratingColors[2];
    case rating > 6.7:
      return ratingColors[3];
    case rating > 6.1:
      return ratingColors[3];
    case rating > 5.5:
      return ratingColors[3];
    default:
      return ratingColors[4];
  }
}

export const slidesToShow = (size) => {
  switch (true) {
    case size.width > 900:
      return 5;
    case size.width > 620:
      return 4;
    case size.width > 470:
      return 3;
    case size.width > 350:
      return 2;
    default:
      return 1;
  }
};
