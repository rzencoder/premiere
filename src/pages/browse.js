import React, { useContext, useEffect, useState } from "react";
import { FirebaseContext } from "../context/firebase";
import { Loader, Navigation, Feature } from "../components";
import Logo from "../logo.png";
import axios from "axios";
import contentData from "../data/content.json";

export default function Browse() {
  const { firebase } = useContext(FirebaseContext);
  const user = firebase.auth().currentUser || {};
  const [loading, setLoading] = useState(true);
  const [openProfile, setOpenProfile] = useState(false);
  const [selectedFeature, setSelectedFeature] = useState(contentData.films[0]);
  const [feature, setFeature] = useState({});
  const [storedFeatureData, setStoredFeatureData] = useState([]);

  // Initial one time useEffect
  useEffect(() => {
    console.log(selectedFeature);

    axios
      .get(`http://www.omdbapi.com/?apikey=12350c73&t=${selectedFeature.title}`)
      .then(({ data }) => {
        setFeature(data);
        setStoredFeatureData([data, ...storedFeatureData]);
      });
    setTimeout(() => {
      setLoading(false);
    }, 2500);
  }, []);

  // Api call when new film chosen
  useEffect(() => {
    const savedFeature = storedFeatureData.filter(
      (item) => item.Title === selectedFeature.title
    );
    console.log(savedFeature);
    if (savedFeature.length !== 0) {
      return setFeature(savedFeature[0]);
    }
    console.log("api");
    axios
      .get(`http://www.omdbapi.com/?apikey=12350c73&t=${selectedFeature.title}`)
      .then(({ data }) => {
        setFeature(data);
        setStoredFeatureData([data, ...storedFeatureData]);
      });
  }, [selectedFeature]);

  if (loading) {
    return (
      <Loader>
        <Loader.Spinner />
      </Loader>
    );
  }
  console.log(feature);
  return (
    <>
      <Feature>
        <Feature.Background src={selectedFeature.slug} />
        <Feature.Overlay />
        <Navigation>
          <Navigation.BrowseLogo src={Logo} alt="Premiere" />

          <Navigation.Section>
            <Navigation.Item>Films</Navigation.Item>
            <Navigation.Item>Series</Navigation.Item>
          </Navigation.Section>
          <Navigation.Search>
            <Navigation.SearchBox></Navigation.SearchBox>
            <Navigation.SearchIcon src="/images/icons/search.png" />
          </Navigation.Search>
          <Navigation.Profile>
            <Navigation.ProfileImage
              onClick={() => setOpenProfile(!openProfile)}
            >
              <Navigation.ProfileImageText>
                {user.displayName[0]}
              </Navigation.ProfileImageText>
            </Navigation.ProfileImage>
            <Navigation.ProfileMenu open={openProfile}>
              <Navigation.Button onClick={() => firebase.auth().signOut()}>
                Sign out
              </Navigation.Button>
            </Navigation.ProfileMenu>
          </Navigation.Profile>
        </Navigation>

        <Feature.TextContainer>
          <Feature.Title>{feature.Title}</Feature.Title>
          <Feature.Details>
            <div>{feature.Year}</div>
            <div>{feature.Rated}</div>
            <div>{feature.imdbRating}</div>
          </Feature.Details>
          <Feature.Text>{feature.Plot}</Feature.Text>
          <Feature.SubText>{feature.Actors}</Feature.SubText>
          <Feature.Button>Play</Feature.Button>
        </Feature.TextContainer>
      </Feature>
      <div></div>
      {contentData["films"].map((item) => {
        return <div onClick={() => setSelectedFeature(item)}>{item.title}</div>;
      })}
    </>
  );
}
