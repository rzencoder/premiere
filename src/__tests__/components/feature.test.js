import React from "react";
import { render } from "@testing-library/react";
import { Feature } from "../../components";
import { displayAgeRating, getRatingColor } from "../../helpers/browse";

const feature = {
  Title: "Inception",
  Year: "2010",
  imdbRating: "8.8",
  Rated: "R",
  Plot: "A team tries to plant a memory into a business man",
  Actors: ["Leonardo DiCaprio", "Ellen Page", "Cillian Murphy"],
};

describe("<Feature />", () => {
  it("renders <Feature /> component", () => {
    const { container, getByText, getByAltText } = render(
      <Feature>
        <Feature.Background src="images/background.png" />
        <Feature.Overlay />
        <Feature.TextContainer>
          <Feature.Title>{feature.Title}</Feature.Title>
          <Feature.Details>
            <Feature.Year>
              {feature.totalSeasons
                ? feature.totalSeasons + " Seasons"
                : feature.Year}
            </Feature.Year>
            <Feature.AgeRating
              src={displayAgeRating(feature.Rated)}
              alt={feature.Rated}
            />
            <Feature.Rating color={getRatingColor(feature.imdbRating)}>
              {feature.imdbRating}
            </Feature.Rating>
          </Feature.Details>
          <Feature.Text>{feature.Plot}</Feature.Text>
          <Feature.SubText>{feature.Actors}</Feature.SubText>
          <Feature.Button onClick={() => {}}>▶{"  "}Watch Now</Feature.Button>
        </Feature.TextContainer>
      </Feature>
    );

    expect(getByText("Inception")).toBeTruthy();
    expect(getByText("2010")).toBeTruthy();
    expect(
      getByText("A team tries to plant a memory into a business man")
    ).toBeTruthy();
    expect(getByText("8.8")).toBeTruthy();
    expect(getByAltText("R").getAttribute("src")).toEqual(
      "../images/icons/15.png"
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders default background when src is blank", () => {
    const { container, getByText } = render(
      <Feature>
        <Feature.Background src="" />
        <Feature.Overlay />
        <Feature.TextContainer>
          <Feature.Title>{feature.Title}</Feature.Title>
          <Feature.Details>
            <Feature.Year>
              {feature.totalSeasons
                ? feature.totalSeasons + " Seasons"
                : feature.Year}
            </Feature.Year>
            <Feature.AgeRating
              src={displayAgeRating(feature.Rated)}
              alt={feature.Rated}
            />
            <Feature.Rating color={getRatingColor(feature.imdbRating)}>
              {feature.imdbRating}
            </Feature.Rating>
          </Feature.Details>
          <Feature.Text>{feature.Plot}</Feature.Text>
          <Feature.SubText>{feature.Actors}</Feature.SubText>
          <Feature.Button onClick={() => {}}>▶{"  "}Watch Now</Feature.Button>
        </Feature.TextContainer>
      </Feature>
    );

    expect(getByText("Inception")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
