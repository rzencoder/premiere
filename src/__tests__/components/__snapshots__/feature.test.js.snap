// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Feature /> renders <Feature /> component 1`] = `
<header
  class="feature__Container-sc-1k5uu41-0 chUkqh"
>
  <div
    class="feature__Background-sc-1k5uu41-1 cHCmCc"
    src="images/background.png"
  />
  <div
    class="feature__Overlay-sc-1k5uu41-2 kQMryO"
  />
  <div
    class="feature__TextContainer-sc-1k5uu41-3 KDKJa"
  >
    <h2
      class="feature__Title-sc-1k5uu41-4 epWCyI"
    >
      Inception
    </h2>
    <div
      class="feature__Details-sc-1k5uu41-7 jeLgNu"
    >
      <div
        class="feature__Year-sc-1k5uu41-12 iLXQYC"
      >
        <div>
          2010
        </div>
      </div>
      <img
        alt="R"
        class="feature__AgeRating-sc-1k5uu41-10 ikNvIH"
        src="../images/icons/15.png"
      />
      <div
        class="feature__Rating-sc-1k5uu41-11 hJZlIe"
        color="#227b08"
      >
        <div>
          8.8
        </div>
      </div>
    </div>
    <p
      class="feature__Text-sc-1k5uu41-5 kYNQDV"
    >
      A team tries to plant a memory into a business man
    </p>
    <div
      class="feature__SubText-sc-1k5uu41-8 cFXPYU"
    >
      Leonardo DiCaprio
      Ellen Page
      Cillian Murphy
    </div>
    <button
      class="feature__Button-sc-1k5uu41-6 jOhDSI"
    >
      ▶
        
      Watch Now
    </button>
  </div>
</header>
`;

exports[`<Feature /> renders default background when src is blank 1`] = `
<header
  class="feature__Container-sc-1k5uu41-0 chUkqh"
>
  <div
    class="feature__Background-sc-1k5uu41-1 hXGHEz"
    src=""
  />
  <div
    class="feature__Overlay-sc-1k5uu41-2 kQMryO"
  />
  <div
    class="feature__TextContainer-sc-1k5uu41-3 KDKJa"
  >
    <h2
      class="feature__Title-sc-1k5uu41-4 epWCyI"
    >
      Inception
    </h2>
    <div
      class="feature__Details-sc-1k5uu41-7 jeLgNu"
    >
      <div
        class="feature__Year-sc-1k5uu41-12 iLXQYC"
      >
        <div>
          2010
        </div>
      </div>
      <img
        alt="R"
        class="feature__AgeRating-sc-1k5uu41-10 ikNvIH"
        src="../images/icons/15.png"
      />
      <div
        class="feature__Rating-sc-1k5uu41-11 hJZlIe"
        color="#227b08"
      >
        <div>
          8.8
        </div>
      </div>
    </div>
    <p
      class="feature__Text-sc-1k5uu41-5 kYNQDV"
    >
      A team tries to plant a memory into a business man
    </p>
    <div
      class="feature__SubText-sc-1k5uu41-8 cFXPYU"
    >
      Leonardo DiCaprio
      Ellen Page
      Cillian Murphy
    </div>
    <button
      class="feature__Button-sc-1k5uu41-6 jOhDSI"
    >
      ▶
        
      Watch Now
    </button>
  </div>
</header>
`;
